replica/database.hh

```diff
@@ -396,13 +396,16 @@ public:
         cache_temperature rate;
         lowres_clock::time_point last_updated;
     };
+
+    void deregister_metrics();
+
 private:
     schema_ptr _schema;
     config _config;
     locator::effective_replication_map_ptr _erm;
     lw_shared_ptr<const storage_options> _storage_opts;
     mutable table_stats _stats;
```

Comment:
There're more metrics that could be deregistered, but they're not causing problems, hence I limited myself only to 2. But I'm leaning more towards removing all of them, for the sake of consistency.